packages/platform-server/src/llm/reducers/callTools.llm.reducer.ts

```
@@ -109,6 +111,10 @@ export class CallToolsLLMReducer extends Reducer<LLMState, LLMContext> {
     const toolsMap = this.createToolsMap();
     const nodeId = ctx?.callerAgent?.getAgentNodeId?.() ?? null;
     const llmEventId = state.meta?.lastLLMEventId ?? null;
```

[major] We now rely on  to push context-count updates after tool execution. If that metadata is ever missing we silently no-op (`LLMCallContextItemCounter` never persists because `eventId` is null) and the LLMCall record will permanently miss all tool-result items. Internal reducers are supposed to fail loudly when invariants are broken, so please guard this path (e.g. throw when `toolsToCall.length > 0` but the event id is absent) instead of continuing with a stale counter.